File: Fresco/FrescoSample/src/main/java/com/yline/fresco/activity/sample/SampleRecyclerActivity.java
package com.yline.fresco.activity.sample;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.graphics.drawable.Animatable;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;

import com.facebook.imagepipeline.image.ImageInfo;
import com.yline.base.BaseAppCompatActivity;
import com.yline.fresco.FrescoManager;
import com.yline.fresco.common.FrescoCallback;
import com.yline.fresco.sample.R;
import com.yline.fresco.view.FrescoView;
import com.yline.test.UrlConstant;
import com.yline.view.recycler.adapter.CommonRecyclerAdapter;
import com.yline.view.recycler.holder.RecyclerViewHolder;

import java.util.ArrayList;
import java.util.List;

public class SampleRecyclerActivity extends BaseAppCompatActivity {

    public static void launcher(Context context) {
        if (null != context) {
            Intent intent = new Intent(context, SampleRecyclerActivity.class);
            if (!(context instanceof Activity)) {
                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
            }
            context.startActivity(intent);
        }
    }

    private RecyclerAdapter mRecyclerAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sample_recycler);

        RecyclerView recyclerView = (RecyclerView) findViewById(R.id.recycler_view);
        recyclerView.setLayoutManager(new GridLayoutManager(this, 3));

        mRecyclerAdapter = new RecyclerAdapter();
        recyclerView.setAdapter(mRecyclerAdapter);

        List<String> resultList = new ArrayList<>();
        for (int i = 0; i < 400; i++) {
            resultList.add(UrlConstant.getGif());
        }
        mRecyclerAdapter.setDataList(resultList);
    }

    private class RecyclerAdapter extends CommonRecyclerAdapter<String> {

        @Override
        public int getItemRes() {
            return R.layout.item_recycler;
        }

        @Override
        public void onBindViewHolder(final RecyclerViewHolder holder, int position) {
            FrescoView frescoView = holder.get(R.id.fresco_view_recycler);
            // FrescoManager.setImageUri(frescoView, sList.get(position));

            FrescoManager.setDynamicUri(frescoView, sList.get(position), false, false, new FrescoCallback.OnSimpleLoadCallback() {
                @Override
                public void onStart(String id, Object callerContext) {

                }

                @Override
                public void onFailure(String id, Throwable throwable) {

                }

                @Override
                public void onSuccess(String id, @Nullable ImageInfo imageInfo, @Nullable Animatable animatable) {
                    if (null != animatable) {
                        if (animatable.isRunning()) {
                            animatable.stop();
                        } else {
                            // animatable.start();
                        }
                    }
                }
            });
        }
    }
}

File: Glide/GildeSample/src/main/java/com/glide/sample/SampleRecyclerActivity.kt
package com.glide.sample

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.widget.ImageView
import androidx.recyclerview.widget.GridLayoutManager
import com.glide.R
import com.glide.sdk.ImageLoader
import com.yline.base.BaseAppCompatActivity
import com.yline.test.UrlConstant
import com.yline.view.recycler.adapter.AbstractRecyclerAdapter
import com.yline.view.recycler.holder.RecyclerViewHolder
import kotlinx.android.synthetic.main.activity_sample_recycler.sample_recycler

/**
 * created on 2020-06-08 -- 20:52
 * @author yline
 */
class SampleRecyclerActivity : BaseAppCompatActivity() {
    companion object {
        fun launch(context: Context?) {
            if (null == context) {
                return
            }

            val intent = Intent()
            intent.setClass(context, SampleRecyclerActivity::class.java)
            if (context !is Activity) {
                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
            }
            context.startActivity(intent)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_sample_recycler)

        val recyclerAdapter = SampleRecyclerAdapter()
        sample_recycler.layoutManager = GridLayoutManager(this, 2)
        sample_recycler.adapter = recyclerAdapter

        val dataList = ArrayList<String>()
        for (i in 0..99) {
            dataList.add(UrlConstant.getUrl())
        }

        recyclerAdapter.setDataList(dataList, true)
    }

    inner class SampleRecyclerAdapter : AbstractRecyclerAdapter<String>() {
        override fun getItemRes(): Int {
            return R.layout.item_sample_recycler
        }

        override fun onBindViewHolder(
            holder: RecyclerViewHolder,
            position: Int
        ) {
            val url = get(position)

            val targetView = holder.get<ImageView>(R.id.item_sample_recycler_img)
            ImageLoader.displayImage(targetView, url)

            holder.setText(R.id.item_sample_recycler_text, url)
        }

    }
}